specification/purviewdatagovernance/PurviewDataCatalog/models.tsp

```diff
@@ -0,0 +1,1291 @@
+model CriticalDataElementsResponse {
+  @doc("The list of critical data elements.")
+  value: CatalogModelCriticalDataElement[];
+
+  @doc("The link to the next page of results.")
+  nextLink?: string;
+}
+
+@doc("Response containing a list of data assets.")
+model DataAssetsResponse {
+  @doc("The list of data assets.")
+  value: CatalogModelDataAsset[];
+
+  @doc("The link to the next page of results.")
+  nextLink?: string;
+}
+
+@doc("Response containing a list of relationship wrappers.")
+model RelationshipWrappersResponse {
+  @doc("The list of relationships.")
+  value: RelationshipWrapper[];
+
+  @doc("The link to the next page of results.")
+  nextLink?: string;
+}
+
+@doc("CatalogModelDataProductTypeEnum")
+union CatalogModelDataProductTypeEnum {
+  @doc("The status is Master.")
+  Master: "Master",
+
+  @doc("The status is Reference.")
+  Reference: "Reference",
+
+  @doc("The status is Analytical.")
+  Analytical: "Analytical",
+
+  @doc("The status is AI.")
+  AI: "AI",
+
+  @doc("The status is MasterDataAndReferenceData.")
+  MasterDataAndReferenceData: "MasterDataAndReferenceData",
+
+  @doc("The status is BusinessSystemOrApplication.")
+  BusinessSystemOrApplication: "BusinessSystemOrApplication",
+
+  @doc("The status is ModelTypes.")
+  ModelTypes: "ModelTypes",
+
+  @doc("The status is DashboardsOrReports.")
+  DashboardsOrReports: "DashboardsOrReports",
+
+  @doc("The status is Operational.")
+  Operational: "Operational",
+
+  @doc("The status is MLAITrainingDataSet.")
+  MLAITrainingDataSet: "MLAITrainingDataSet",
+
+  @doc("The status is MLAITestingDataSet.")
+  MLAITestingDataSet: "MLAITestingDataSet",
+
+  @doc("The status is TransactionalDataset.")
+  TransactionalDataset: "TransactionalDataset",
+
+  @doc("The status is AnalyticsModel.")
+  AnalyticsModel: "AnalyticsModel",
+
+  @doc("The status is SemanticModel.")
+  SemanticModel: "SemanticModel",
+
+  string,
+}
+
+@doc("UpdateFrequencyEnum")
+union UpdateFrequencyEnum {
+  @doc("The status is Hourly.")
+  Hourly: "Hourly",
+
+  @doc("The status is Daily.")
+  Daily: "Daily",
+
+  @doc("The status is Weekly.")
+  Weekly: "Weekly",
+
+  @doc("The status is Monthly.")
+  Monthly: "Monthly",
+
+  @doc("The status is Quarterly.")
+  Quarterly: "Quarterly",
+
+  @doc("The status is Yearly.")
+  Yearly: "Yearly",
+
+  string,
+}
+
+@doc("Microsoft Purview Data Governance Catalog Model Data Product.")
+model CatalogModelDataProduct {
+  @doc("The status of the data product.")
+  status?: CatalogModelStatus;
+
+  @doc("The type of the data product.")
+  type?: CatalogModelDataProductTypeEnum;
+
+  @doc("The update frequency of the data product.")
+  updateFrequency?: UpdateFrequencyEnum;
+
+  @doc("The managed attributes associated with the data product.")
+  managedAttributes?: CatalogModelManagedAttribute[];
+
+  @doc("The unique identifier of the data product.")
+  id?: uuid;
+
+  @doc("The name of the data product.")
+  name?: string;
```

Comment:
Why every property in this model is optional? It can have no `id`, no `name`, no `status`?